### PR TITLE
Adds API to get all values in collection

### PR DESCRIPTION
**Pull request by DarthMike**

### CODE REVIEW

File: Pod/Common/YapDatabaseExtensions.swift [hunk 1]
```diff
@@ -240,6 +244,10 @@ extension YapDatabaseConnection {
         return read(readInTransactionValuesAtIndexes(indexes))
     }
 
+    public func readAllValuesInCollection<V where V: Saveable, V: Persistable, V.ArchiverType.ValueType == V>(collection: String) -> [V] {
+        return self.read { $0.readAllValuesInCollection(collection) }
```
Comment (danthorpe) [inline Pod/Common/YapDatabaseExtensions.swift]: Very nice, but can you remove these `self.`s please?

File: Pod/Common/YapDatabaseExtensions.swift [hunk 2]
```diff
@@ -442,6 +450,11 @@ extension YapDatabaseReadTransaction {
     public func readValuesAtIndexes<V where V: Saveable, V: Persistable, V.ArchiverType.ValueType == V>(indexes: [YapDatabase.Index]) -> [V] {
         return indexes.mapOptionals(readValueAtIndex)
     }
+    
+    public func readAllValuesInCollection<V where V: Saveable, V: Persistable, V.ArchiverType.ValueType == V>(collection: String) -> [V] {
+        let keys = self.allKeysInCollection(collection) as? [String]
+        return self.readValuesForKeys(keys ?? [])
```
Comment (danthorpe) [inline Pod/Common/YapDatabaseExtensions.swift]: Same here, thanks!

**Comment by danthorpe** [comment]
Awesome @DarthMike, will merge it. Feel free to make issues or PRs for any APIs you might think are useful.

**Comment by DarthMike** [comment]
@danthorpe Removed usage of self :rocket:

**Pull request merged**